From 2cc62644f24947e50a6ce17a6c685e4a3af49af3 Mon Sep 17 00:00:00 2001
Upstream-Status: Rejected, Fixed differently [https://github.com/sandstorm-io/capnproto/pull/210]
From: Clemens Lang <neverpanic@gmail.com>
Date: Sat, 23 May 2015 19:02:32 +0200
Subject: [PATCH] Avoid incorrect library version with BSD sed

BSD sed does not support GNU sed's -r switch, but uses -E instead for
the same functionality. Because -r was used in Makefile.am to set the
SO_VERSION and failed there on platforms with BSD sed (e.g. OS X and
possibly some BSDs, too) SO_VERSION was empty, causing libtool to use
the next given parameter as version number, which was -no-undefined.

Fix this by testing for the regex flag in configure.ac (note that some
implementations of sed support -E but do not document it).

Signed-off-by: Clemens Lang <neverpanic@gmail.com>
---
 c++/Makefile.am  |  2 +-
 c++/configure.ac | 15 +++++++++++++++
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/c++/Makefile.am b/c++/Makefile.am
index 8e0d9dd..b149b78 100644
--- a/c++/Makefile.am
+++ b/c++/Makefile.am
@@ -189,7 +189,7 @@ endif
 
 # Don't include security release in soname -- we want to replace old binaries
 # in this case.
-SO_VERSION = $(shell echo $(VERSION) | sed -re 's/^([0-9]+[.][0-9]+[.][0-9]+)([.][0-9]+)?(-.*)?$$/\1\3/g')
+SO_VERSION = $(shell echo $(VERSION) | $(ESED) -e 's/^([0-9]+[.][0-9]+[.][0-9]+)([.][0-9]+)?(-.*)?$$/\1\3/g')
 
 libkj_la_LIBADD = $(PTHREAD_LIBS)
 libkj_la_LDFLAGS = -release $(SO_VERSION) -no-undefined
diff --git a/c++/configure.ac b/c++/configure.ac
index 26d9dcf..fe579c8 100644
--- a/c++/configure.ac
+++ b/c++/configure.ac
@@ -52,6 +52,21 @@ AC_PROG_CXX
 AC_LANG([C++])
 AX_CXX_COMPILE_STDCXX_11
 
+AC_MSG_CHECKING([how to run sed with extended regex support])
+AS_IF(
+    [test "$(echo test | sed -E 's/([st]+)$/xx\1/' 2>/dev/null)" = "texxst"], [
+        AC_MSG_RESULT([sed -E])
+        ESED="sed -E"
+    ],
+    [test "$(echo test | sed -r 's/([st]+)$/xx\1/' 2>/dev/null)" = "texxst"], [
+        AC_MSG_RESULT([sed -r])
+        ESED="sed -r"
+    ],
+    [AC_MSG_ERROR([cannot find sed with extended regex support (neither -r nor -E work)])]
+)
+AC_SUBST([ESED])
+
+
 AS_CASE("${host_os}", *mingw*, [
     # We don't use pthreads on MinGW.
     PTHREAD_CFLAGS="-mthreads"
-- 
2.4.1

